dhcpv6/options.go

```diff
@@ -208,10 +208,12 @@ func (o *Options) FromBytesWithParser(data []byte, parser OptionParser) error {
 		// Consume, but do not Copy. Each parser will make a copy of
 		// pertinent data.
 		optData := buf.Consume(length)
-
+		if err := buf.Error(); err != nil {
```

Comment:
same, lets discuss error messages on a separate PR and in this one focus on ZTP which if I am not mistake is the one more pressing for you